lms/djangoapps/learner_dashboard/permissions.py

```diff
@@ -0,0 +1,32 @@
+"""
+Permissions for program discussion api
+"""
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import permissions, status
+from rest_framework.exceptions import APIException
+
+from lms.djangoapps.program_enrollments.api import get_program_enrollment
+
+
+class IsEnrolledInProgram(permissions.BasePermission):
+    """Permission that checks to see if the user is enrolled in the course or is staff."""
+    def has_permission(self, request, view):
+
+        """Returns true if the user is enrolled in program"""
+        if not view.program:
```

Comment:
what is `view.program` value here? and where the value is being passed?

Reply:
`view` is passed in `has_permission` function  by default and `program` attribute comes from `ProgramSpecificViewMixin` from which view is subclassed.

Reply:
got it. thanks.